### ISSUE TITLE
stat/distmv - cumulative distribution function for multivariate normal distribution

### ISSUE THREAD

**Issue by JKhawaja** [gonum/gonum#1007]
### What are you trying to do?

A cumulative distribution function is not currently available for multivariate normal distributions. It would be useful to have one. One exists for univariate normal distributions.

### Is this feature absent from the current master?

Yes.

### Are you able to help contribute to the feature?

Possibly.

**Comment by sbinet** [comment]
that's probably in @btracey 's area of expertise.

**Comment by JKhawaja** [comment]
Proposal:

Taken from paper: https://upload.wikimedia.org/wikipedia/commons/a/a2/Cumulative_function_n_dimensional_Gaussians_12.2013.pdf

```
r := Mahalanobis distance (of input to mean vector)
n := dimension of distribution
if n is even:
        m := n/2
        numerator := integral{x \in 0,r}(x^n-1 * e^(-(x^2)/2))
        denominator := factorial(2^m-1 * (m-1))
        return numerator/denominator
if n is odd:
        numerator := integral{x \in 0,r}(x^n-1 * e^(-(x^2)/2))
        denominator := (product of all odd natural numbers up to and including n-2) * squrt(2*pi)
        return numerator/denominator
```

Unverified.

**Comment by btracey** [comment]
I don't think that algorithm is computing the same object. This is computing a one-dimensional output, while the function for Uniform computes the multi-dimensional CDF ( https://en.wikipedia.org/wiki/Cumulative_distribution_function#Multivariate_case ). It could be added if that's what you're interested in (under a different name), but it's not the same thing.

**Comment by JKhawaja** [comment]
@btracey  Ah, didn't know that, very cool! You're the expert. But, I think it would be interesting to have both.

From the paper (for the single-dimensional result):

> Now, our aim is to find the cumulative function of the Gaussian which composes X as a function of the Mahalanobis distance r, that is, to find for each r, the proportion of instances of X that falls statistically at a Mahalanobis distance r′ ≤ r from the Gaussian composing X

So, I have no idea what a good name for it would be ... but perhaps `MahalanobisCDF`? hehe

**Comment by sdressler** [comment]
Hi,

Is there possibly an update to this? I'd need the CDF for this MV distribution as well. As a workaround, I used `(*Normal) MarginalNormalSingle` to get the UV marginals and use their respective CDFs. However, I am not sure whether this is a feasible approach.

Thanks,
Sebastian
